src/connection.js

```diff
@@ -547,12 +540,23 @@ function Connection(options, queues = {}, { onopen = noop, onend = noop, onclose
 
       if (needsTypes) {
         initial.reserve && (initial = null)
-        return fetchArrayTypes()
+
+        // Prevent duplicate requests
+        needsTypes = false
+
+        // Store the promise so concurrent requests can wait for type initialization to complete
+        typesPromise = fetchArrayTypes().finally(() => {
```

Comment:
if this promise rejects, it throws a global promise rejection. Is there anyway to pipe through an onError handler for new connections or will this PR fix that issue as well and i am misreading the code? 

Again thank you for this fix, its very much needed by our team